lib/kubernetes-utils.js

```diff
@@ -80,13 +80,19 @@ function substituteImageNameInSpecContent(currentString, imageName, imageNameWit
         return currentString;
     }
     return currentString.split('\n').reduce((acc, line) => {
-        const imageKeyword = line.match(/^ *image:/);
+        const imageKeyword = line.match(/^ *(- )?image:/);
         if (imageKeyword) {
-            const [currentImageName, currentImageTag] = line
+            let currentImageName = line
                 .substring(imageKeyword[0].length) // consume the line from keyword onwards
                 .trim()
-                .replace(/[',"]/g, '') // replace allowed quotes with nothing
-                .split(':');
+                .replace(/[',"]/g, '');
+
+            const registrySplitIndex = currentImageName.indexOf('/');
+            const lastTagSplitIndex = currentImageName.lastIndexOf(':');
+            if(lastTagSplitIndex > registrySplitIndex) {
+                currentImageName = currentImageName.substring(0, lastTagSplitIndex)
+            }
+
             if (currentImageName === imageName) {
                 return acc + `${imageKeyword[0]} ${imageNameWithNewTag}\n`;
             }

```

Comment:
Instead of this, you could consider parsing the yaml to `json` and using `jsonpath` you could update the image fields